File: src/main/java/com/googongill/aditory/common/ErrorResponse.java
package com.googongill.aditory.common;

import com.googongill.aditory.common.code.BusinessErrorCode;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
@Builder
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class ErrorResponse {

    private final HttpStatus httpStatus;
    private final String message;
    private final boolean success;


    public static ErrorResponse of(BusinessErrorCode businessErrorCode) {
        return ErrorResponse.builder()
                .httpStatus(businessErrorCode.getHttpStatus())
                .message(businessErrorCode.getMessage())
                .success(false)
                .build();
    }
}

File: src/main/java/com/googongill/aditory/common/ErrorResponse.kt
package com.googongill.aditory.common

import com.googongill.aditory.common.code.BusinessErrorCode
import org.springframework.http.HttpStatus

data class ErrorResponse(
    val httpStatus: HttpStatus,
    val message: String,
    val success: Boolean = false
) {

    companion object {
        fun of(businessErrorCode: BusinessErrorCode): ErrorResponse {
            return ErrorResponse(
                httpStatus = businessErrorCode.httpStatus,
                message = businessErrorCode.message
            )
        }
    }

}
